### PR TITLE
Set up secrets

### PR DESCRIPTION
**Pull request by mqharris**
I access the api key using a helper function.

The helper function will check the R environment variables for a key named "VRD_API". If the variable doesn't exist it will remind to user to set it and how to do that.

Im open to suggestions on set up, file names, function names, etc.

`Check()` also gives the following notes:
```
> checking DESCRIPTION meta-information ... NOTE
  Malformed Description field: should contain one or more complete sentences.

> checking top-level files ... NOTE
  Non-standard file/directory found at top level:
    ‘CODE_OF_CONDUCT.md’

> checking R code for possible problems ... NOTE
  get_recall_by_number: no visible global function definition for ‘GET’
  get_recall_by_number: no visible global function definition for
    ‘add_headers’
  get_recall_by_number: no visible global function definition for
    ‘content’
  Undefined global functions or variables:
    GET add_headers content
```

Any ideas?

### CODE REVIEW

File: R/getRecallByNumber.R [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# where does this go?
```
Comment (mqharris) [inline R/getRecallByNumber.R]: 👀

Comment (mqharris) [reply R/getRecallByNumber.R]: Where do imports like `library("whatever")` go

Comment (WraySmith) [reply R/getRecallByNumber.R]: Section 2.16 of the R Packages book shows how to attach a package which will add it into the DESCRIPTION file: `use_package("<packagename>")`. Sections 6.4/6.5 also has a bit of details on this as well. When you want to call the package, just use `<packagename>::<funcname>(<args>)`

You can also check what packages have already been attached by checking the DESCRIPTION file.

**Comment by WraySmith** [comment]
A couple of general comments:
- all functions that use a package (such as httr) should be of the form:  `<packagename>::<funcname>(<args>)`
- recommend being slightly more verbose with variable names such as url instead of u, response instead of r, and content instead of c (I think the following the variable naming conventions of the httr docs is a good idea)

**Comment by WraySmith** [comment]
The API call function `get_recall_by_number` is pretty bare bones as it doesn't include error handling etc, recommend it follows the example in the article: https://httr.r-lib.org/articles/api-packages.html including:
- setting a user agent (use the github address
- error handling
- parsing
- providing an S3 output class

Looking through a few other repos, this httr form seems pretty standard. Or, if you were planning on having Ryan work on this as part of his functions, that works too and can just merge without addressing. Would also recommend looking at a few other API wrapper packages and checking that we're not missing key functionality.

**Comment by WraySmith** [comment]
The filename getRecallByNumber is pretty long, also need to decide if we're using camelCase or snake_case. Here's a link to a couple style notes for R: http://adv-r.had.co.nz/Style.html. Maybe a file name like recall_by_number.R?

Something like this should probably be defined in the contributions.md document - @RyKoe, where are we at with this document?

**Comment by RyKoe** [comment]
> The filename getRecallByNumber is pretty long, also need to decide if we're using camelCase or snake_case. Here's a link to a couple style notes for R: http://adv-r.had.co.nz/Style.html. Maybe a file name like recall_by_number.R?
> 
> Something like this should probably be defined in the contributions.md document - @RyKoe, where are we at with this document?

I'll be working on this project all day today.  The contributions page I'll have up momentarily.

File: R/utils.R [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+#'
+#' @examples
+#' get_vrd_api()
+get_vrd_api <- function() {
+    x <- Sys.getenv("VRD_API")
+    if (x == "") {
+        stop("You must set your VRD_API with `Sys.setenv(VRD_API = 'your_api_key_here')`")
+    }
+    return(x)
+}
```
Comment (WraySmith) [inline R/utils.R:14]: Recommend following the function form used in the httr docs:
```
github_pat <- function() {
  pat <- Sys.getenv('GITHUB_PAT')
  if (identical(pat, "")) {
    stop("Please set env var GITHUB_PAT to your github personal access token",
      call. = FALSE)
  }

  pat
}
```

**Comment by WraySmith** [comment]
Another general comment is to try and follow the example code a bit more directly upfront as is possible to avoid going back and needing to modify based on that.

**Pull request merged**